distributed/worker_state_machine.py

```diff
@@ -1468,17 +1473,23 @@ def _purge_state(self, ts: TaskState) -> None:
         self.long_running.discard(ts)
         self.in_flight_tasks.discard(ts)
 
+    def _should_throttle_incoming_transfers(self) -> bool:
```

Comment:
I recommend to write a short doc string describing in words what the conditions below are expressing.
Particularly with the long names and weird formatting, readability is not great (not your fault)